source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -0,0 +1,200 @@
+#include "CreateMaterialXNodeCmd.h"
+#include "MaterialXNode.h"
+
+#include <MaterialXCore/Document.h>
+#include <MaterialXFormat/XmlIo.h>
+
+#include <maya/MArgParser.h>
+#include <maya/MSelectionList.h>
+#include <maya/MFnDependencyNode.h>
+#include <maya/MSyntax.h>
+#include <maya/MArgDatabase.h>
+#include <maya/MStringArray.h>
+#include <maya/MFnNumericAttribute.h>
+#include <maya/MGlobal.h>
+
+#include <algorithm>
+#include <sstream>
+
+#define kDocumentFlag     "d"
+#define kDocumentFlagLong "document"
+
+#define kTargetFlag       "t"
+#define kTargetFlagLong   "target"
+
+CreateMaterialXNodeCmd::CreateMaterialXNodeCmd()
+{
+}
+
+CreateMaterialXNodeCmd::~CreateMaterialXNodeCmd()
+{
+}
+
+MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
```

Comment:
Creates a MaterialXNode and adds attributes to it based upon the MaterialX document inputs/parameters for the provided node.